tooling/noir_js/src/index.ts

```diff
@@ -1,8 +1,59 @@
-import * as acvm from '@noir-lang/acvm_js';
-import * as abi from '@noir-lang/noirc_abi';
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+import { Backend, CompiledCircuit, ProofData } from '@noir-lang/types';
+import { generateWitness } from './witness_generation.js';
+import initAbi, * as abi from '@noir-lang/noirc_abi';
+import initACVM, * as acvm from '@noir-lang/acvm_js';
+import { witnessMapToUint8Array } from './serialize.js';
 
-export { acvm, abi };
+const { abiDecode } = abi;
 
-export { acirToUint8Array, witnessMapToUint8Array } from './serialize.js';
+class Noir {
+  constructor(private backend?: Backend) {}
 
-export { Noir } from './program.js';
+  async init(): Promise<void> {
+    // If these are available, then we are in the
+    // web environment. For the node environment, this
+    // is a no-op.
+    if (typeof initAbi === 'function') {
+      await Promise.all([initAbi(), initACVM()]);
+    }
+  }
+
+  async destroy(): Promise<void> {
+    if (!this.backend) throw new Error('No backend to destroy');
+
+    await this.backend.destroy();
+  }
+
+  private getBackend(): Backend {
+    if (this.backend === undefined) throw new Error('Operation requires a backend but none was provided');
+    return this.backend;
+  }
+
+  // Initial inputs to your program
+  async execute(
+    inputs: abi.InputMap,
+    circuit?: CompiledCircuit,
```

Comment:
I think it's a better API to instantiate the `Noir` class with a `CompiledCircuit` and use that in this function rather than having to pass in a circuit each time.

This is inconsistent when `Noir` "knows" about the circuit you're interacting with when you're proving/verifying but not for execution.